ma_gym/envs/lumberjacks/lumberjacks.py

```diff
@@ -0,0 +1,351 @@
+
+        Args:
+            mask: tuple of slices in extended coordinates.
+        """
+        agent_iter = np.ndenumerate(np.sum(self._agent_map[mask], axis=2))
+        tree_iter = np.nditer(self._tree_map[mask])
+        for (pos, n_a), n_t in zip(agent_iter, tree_iter):
+            yield pos, n_a, n_t
+
+    def get_agent_obs(self) -> List[List[float]]:
+        """Returns list of observations for each agent."""
+        obs = np.zeros((self.n_agents, self._obs_len))
+        for i, (agent_id, agent) in enumerate(self._agent_generator()):
+            rel_pos = self._to_relative_coordinates(agent.pos)
+            obs[i, 0] = agent_id / self.n_agents  # Agent ID
+            obs[i, 1] = rel_pos[0] / (self._grid_shape[0] - 1)  # Coordinate
+            obs[i, 2] = rel_pos[1] / (self._grid_shape[1] - 1)  # Coordinate
+            obs[i, 3] = self._step_count / self._max_steps  # Steps
+
+            for j, (_, agent_strength, tree_strength) in zip(
+                    itertools.count(start=4, step=2),
+                    self._agent_view_generator(agent.pos, self._agent_view)):
+                obs[i, j] = agent_strength / self.n_agents
+                obs[i, j+1] = tree_strength / self.n_agents
+
+        # Convert it from numpy array
+        obs = obs.tolist()
+
+        if self.full_observable:
+            obs = [feature for agent_obs in obs for feature in agent_obs]
+            obs = [obs] * self.n_agents
+
+        return obs
+
+    def _agent_generator(self) -> Tuple[int, Agent]:
+        """Yields agent_id and agent for all agents in environment."""
+        for agent_id, agent in enumerate(self._agents):
+            yield agent_id, agent
+
+    def _agent_view_generator(self, pos: Coordinates, view_range: Tuple[int, int]):
+        """Yields position, number of agent and tree strength for cells in distance of `view_range` from `pos`.  """
+        mask = (
+            slice(pos[0] - view_range[0], pos[0] + view_range[0] + 1),
+            slice(pos[1] - view_range[1], pos[1] + view_range[1] + 1),
+        )
+        yield from self._view_generator(mask)
+
+    def step(self, agents_action: List[int]):
+        assert len(agents_action) == self.n_agents
+
+        self._step_count += 1
+        rewards = np.full(self.n_agents, self._step_cost)
+
+        # Move agents
+        for (agent_id, agent), action in zip(self._agent_generator(), agents_action):
+            if not self._agent_dones[agent_id]:
+                self._update_agent_pos(agent, action)
+
+        # Cut down trees
+        mask = (np.sum(self._agent_map, axis=2) >= self._tree_map) & (self._tree_map > 0)
+        self._tree_map[mask] = 0
+
+        # Calculate rewards
+        rewards += np.sum(mask * self._tree_cutdown_reward, axis=(0, 1))
+        self._total_episode_reward += rewards
+
+        if (self._step_count >= self._max_steps) or (np.count_nonzero(self._tree_map) == 0):
+            self._agent_dones = [True] * self.n_agents
+
```

Comment:
Add warning if user is still stepping after all agents are done. 

example: https://github.com/koulanurag/ma-gym/blob/master/ma_gym/envs/checkers/checkers.py#L202